lib/common/socket.c

```diff
@@ -798,12 +800,16 @@ static int on_async_resumption_new(SSL *ssl, SSL_SESSION *session)
     i2d_SSL_SESSION(session, &p);
 
     id = SSL_SESSION_get_id(session, &id_len);
-    resumption_new(h2o_iovec_init(id, id_len), data);
+    resumption_new(sock, h2o_iovec_init(id, id_len), data);
     return 0;
 }
 
 static void on_async_resumption_remove(SSL_CTX *ssl_ctx, SSL_SESSION *session)
 {
+    if (resumption_remove == NULL) {
+        return;
+    }
```

Comment:
> note also that we would not want to remove a cache entry when a connection is abruptly shut down, which is often the case with the browsers

ah yes, this was the second reason. thank you for the complement.

> I would appreciate it if you create a PR against master that fixes the issue.
I'll do it